output/baseoutput/clientsession.go

```diff
@@ -217,17 +231,18 @@ func (session *clientSession) collectLeftovers(maybePreviousLeftovers chan base.
 		session.logger.Panic("invalid ending type: ", ending)
 	}
 
-	if !session.ackerEnded.Wait(defs.ForwarderAckerStopTimeout) {
-		session.logger.Errorf("BUG: timeout waiting for acknowledger to stop. stack=%s", util.Stack())
+	if !session.ackerEnded.Wait(defs.IntermediateChannelTimeout) {
+		session.logger.Errorf("BUG: timeout waiting for acknowledger to hard stop. stack=%s", util.Stack())
 	}
 	fromAckerChannel := collectChunksFromChannel(session.ackerChan)
 
 	// gather pendings chunks left in runAcknowledger's pendingChunksByID
 	var fromAckerPending []base.LogChunk
-	if pendingListPtr := session.unacked.Load().(*[]base.LogChunk); pendingListPtr != nil {
-		fromAckerPending = *pendingListPtr
+	pendingListPtr, ok := session.unacked.Load().(*[]base.LogChunk)
```

Comment:
`unacked` and other usages of Pointer can be simplified with `util.AtomicRef<T>` with generics, and maybe add null check there